tests/simple/testdata/optionals.textproto

```diff
@@ -234,4 +234,113 @@ section: {
       }
     }
   }
+  test {
+    name: "optional_empty_map_optindex_hasValue"
+    expr: "optional.of({}).?c.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "empty_struct_optindex_hasValue"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{}.?repeated_string.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "optional_empty_struct_optindex_hasValue"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "optional.of(TestAllTypes{}).?repeated_string.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "optional_none_optselect_hasValue"
+    expr: "optional.none().?repeated_string.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "struct_optindex_value"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{repeated_string: ['foo']}.?repeated_string.value()"
+    value: {
+      list_value: {
+        values {
+          string_value: "foo"
+        }
+      }
+    }
+  }
+  test {
+    name: "optional_struct_optindex_value"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "optional.of(TestAllTypes{repeated_string: ['foo']}).?repeated_string.value()"
```

Comment:
Can you add a variant of this that attempts to get the element at `repeated_string[0].value()` as well?

Reply:
Done.